app/commands/generate_missing_mappings_command.rb

```diff
@@ -1,68 +1,116 @@
 # frozen_string_literal: true
 
 class GenerateMissingMappingsCommand < ApplicationCommand
-  MAX_RETRIES = 3
-  QDRANT_PORT = 6333
+  UNSTABLE = "unstable"
   EMBEDDING_MODEL = "text-embedding-3-small"
   MAPPING_GRADER_GPT_MODEL = "gpt-4"
 
   usage do
     no_command
   end
 
+  environment :openai_api_key do
+    desc "OpenAI API key for mappings generation"
+    required
+  end
+
+  environment :openai_url do
+    desc "OpenAI API URL for mappings generation"
+    default "https://openai-proxy.shopify.ai/v1" # TODO: we should probably not have this in this repo
+  end
+
+  environment :qdrant_url do
+    desc "Qdrant API URL for embeddings search"
+    default "http://localhost:6333"
+  end
+
+  option :shopify_version do
+    desc "Target shopify taxonomy version"
+    short "-V"
+    long "--version string"
+    default UNSTABLE
+  end
+
+  option :retries do
+    desc "Number of retries for OpenAI API"
+    short "-r"
+    long "--retries integer"
+    default 3
+    convert :to_i
+  end
+
   def execute
+    @embeddings = {}
```

Comment:
Was a thought on if this could save passing certain variables around all the time... but its unfinished so might not be a keeper

Reply:
I thought about making embeddings an instance variable, but they are specific to each channel integration (though, at the moment, there's only one: Google)